usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeekTrainingManagerClient.java

```diff
@@ -0,0 +1,199 @@
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] weekTrainings = jsonArray.split(BRACKET_SLASH);
+            weekTrainingList.add(GSON.fromJson(weekTrainings[0], WeekTraining.class));
+            for (int i = 1; i < weekTrainings.length; i++) {
+                weekTrainings[i] = BRACKET + weekTrainings[i];
+                weekTrainingList.add(GSON.fromJson(weekTrainings[i], WeekTraining.class));
+            }
+        }
+        return weekTrainingList;
+    }
+
+    /**
+     * Gets the data from all the weekTrainings of the pet between the initial and final
+     * date not including them.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @param initialDate Initial Date
+     * @param finalDate Final Date
+     * @return The List containing all the weekTrainings eaten by the pet in the specified time
+     * @throws ExecutionException When the deletion fails
+     * @throws InterruptedException When the deletion is interrupted
+     */
+    public List<WeekTraining> getAllWeekTrainingsBetween(String accessToken, String owner,
+                                                         String petName,
+                                             DateTime initialDate, DateTime finalDate)
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + SLASH + petName
+                + "/between/" + initialDate + SLASH + finalDate, accessToken).get();
+        List<WeekTraining> weekTrainingList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] weekTrainings = jsonArray.split(BRACKET_SLASH);
+            weekTrainingList.add(GSON.fromJson(weekTrainings[0], WeekTraining.class));
+            for (int i = 1; i < weekTrainings.length; i++) {
+                weekTrainings[i] = BRACKET + weekTrainings[i];
+                weekTrainingList.add(GSON.fromJson(weekTrainings[i], WeekTraining.class));
+            }
+        }
+        return weekTrainingList;
+    }
+
+    /**
+     * Updates the weekTraining's value.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date the weekTraining was created
+     * @param value Value the weekTraining will have
+     * @return The response code
+     * @throws ExecutionException When the update fails
+     * @throws InterruptedException When the update is interrupted
+     */
+    public int updateWeekTrainingField(String accessToken, String owner, String petName,
+                                       DateTime date, Double value)
```

Comment:
Maybe you should consider not using Double. Use double instead. Double is a wrapper class. Here you don't need to use any wrapper.
What does it mean to be a wrapper? I really don't know. But I know you don't need to use it here.